sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4126,7 +4126,8 @@ piEnqueueMemBufferMap(pi_queue Queue, pi_mem Buffer, pi_bool BlockingMap,
     piEventsWait(NumEventsInWaitList, EventWaitList);
     if (Buffer->MapHostPtr) {
       *RetMap = Buffer->MapHostPtr + Offset;
-      memcpy(*RetMap, pi_cast<char *>(Buffer->getZeHandle()) + Offset, Size);
+      if (!(MapFlags & CL_MAP_WRITE_INVALIDATE_REGION))
```

Comment:
Please create a new PI flag instead of using OpenCL directly here.
Also where is this flag being set?

Reply:
And create a PI enumeration of supported flags instead of `cl_map_flags`

Reply:
The flag is set in [memory_manager.cpp:613](https://github.com/intel/llvm/blob/566c136850ac00313b0e74d21898c19f128cbd32/sycl/source/detail/memory_manager.cpp#L613)  - not part of this PR.  


Those `MapFlags`  being passed to the PI routines are of type `cl_map_flag` and both CUDA and Level Zero PI already check for various CL map_flags today (eg  `CL_MAP_WRITE`, `CL_MAP_READ`) in a dozen locations.   There is a TODO recorded in several places to go through and replace `cl_map_flags` with a proper PI constant.    I'm happy to do it, but I don't think it should be part of this PR.   
Is it ok to leave `CL_MAP_WRITE_INVALIDATE_REGION` as it is here for this PR? I will make a ticket for changing cl_map_flags and undertake it afterwards.

Reply:
Sure, let's do this separately.